Dockerfile

```diff
@@ -13,7 +13,7 @@
 
 # ----------- deps -----------
 # Install from Debian Stretch with modern Python support
-FROM python:3.8-slim-buster as deps
+FROM python:3.10-slim-buster as deps
```

Comment:
we already implicitly require `python 3.10` by way of microcosm>4.0 using type annotations of form `type | type` which are only supported on python 3.10 and up